spec/integration/worksheet.spec.js

```diff
@@ -659,6 +659,48 @@ describe('Worksheet', () => {
       });
     });
 
+    context('when the worksheet name contains illegal characters', () => {
+      it('throws an error', () => {
+        const workbook = new ExcelJS.Workbook();
+
+        const invalidCharacters = ['*', '?', ':', '/', '\\', '[', ']'];
+        const expectedErrors = invalidCharacters.map(
+          invalidCharacter =>
+            `Worksheet name ${invalidCharacter} cannot include any of the following characters: * ? : \\ / [ ]`
+        );
+
+        for (
+          let lastIndex = invalidCharacters.length - 1;
+          lastIndex >= 0;
+          --lastIndex
+        ) {
+          expect(() =>
+            workbook.addWorksheet(invalidCharacters[lastIndex])
+          ).to.throw(expectedErrors[lastIndex]);
+        }
+      });
+
+      it('throws an error', () => {
+        const workbook = new ExcelJS.Workbook();
+
+        const invalidNames = ['\'sheetName', 'sheetName\''];
+        const expectedErrors = invalidNames.map(
```

Comment:
I think the array creation and the for-index-loop distract a little bit from what is being tested, I think we should simplify to this:
```js
for (const invalidName of invalidNames) {
  expect(() => workbook.addWorksheet(invalidName)).to.throw(
    `The first or last character of worksheet name cannot be a single quotation mark: ${invalidName}`
  );
}
```